os/mm/mm_heap/mm_check_heap_corruption.c

```diff
@@ -77,7 +77,7 @@ enum node_type_e {
 };
 typedef enum node_type_e node_type_t;
 
-#if defined(__KERNEL__)
+#if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
```

Comment:
Please add this at line 146 as well

Reply:
@thapav Oops sorry, I merged already.
@swasthik22 Could you check @thapav 's comment?

Reply:
@sunghan-chang @thapav sorry I had forgot to push the changes for this, will be raising new PR addressing this comment.